src/EnergyPlus/Material.cc

```diff
@@ -2756,6 +2758,99 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
+                        format("{}: invalid {} entered={}, must match to a valid Material name.",
+                               cCurrentModuleObject,
+                               state.dataIPShortCut->cAlphaFieldNames(1),
+                               alphas(2)));
+        errorsFound = true;
+        return;
+    }
+    auto *thisMaterial = state.dataMaterial->Material(MaterNum);
+
+    if (thisMaterial->Group != Material::MaterialGroup::RegularMaterial) {
+        ShowSevereError(
+            state,
+            format("{}: Reference Material is not appropriate type for Thermal/Solar Absorptance properties, material={}, must have regular "
+                   "properties (Thermal/Solar Absorptance)",
+                   cCurrentModuleObject,
+                   thisMaterial->Name));
+        errorsFound = true;
+        return;
+    }
+
+    Material::VariableAbsCtrlSignal controlSignal = Material::VariableAbsCtrlSignal::SurfaceTemperature; // default value
+    controlSignal = static_cast<VariableAbsCtrlSignal>(getEnumerationValue(VariableAbsCtrlSignalUC, UtilityRoutines::MakeUPPERCase(alphas(3))));
+    //    init to 0 as GetScheduleIndex returns 0 for not-found schedule or curves
+    int functionIdxThermal = 0;
+    int scheduleIdxThermal = 0;
+    int functionIdxSolar = 0;
+    int scheduleIdxSolar = 0;
+    if (controlSignal == VariableAbsCtrlSignal::Scheduled) {
+        scheduleIdxThermal = ScheduleManager::GetScheduleIndex(state, alphas(5));
+        scheduleIdxSolar = ScheduleManager::GetScheduleIndex(state, alphas(7));
+        if (scheduleIdxThermal + scheduleIdxSolar == 0) {
+            ShowSevereError(
+                state,
+                format("{}: Control signal \"Scheduled\" is chosen but both thermal and solar absorptance schedules are undefined, for object {}",
+                       cCurrentModuleObject,
+                       alphas(1)));
+            errorsFound = true;
+            return;
+        }
+    } else { // controlled by performance table or curve
+        functionIdxThermal = Curve::GetCurveIndex(state, alphas(4));
+        functionIdxSolar = Curve::GetCurveIndex(state, alphas(6));
+        if (functionIdxThermal + functionIdxSolar == 0) {
+            format("{}: non-schedule control signal is chosen but both thermal and solar absorptance table or curve are undefined, for object {}",
+                   cCurrentModuleObject,
+                   alphas(1));
+            errorsFound = true;
+            return;
+        }
+    }
+    thisMaterial->variableAbsorptanceCtrlSignal = controlSignal;
+    thisMaterial->variableThermalAbsporptanceScheduleIdx = scheduleIdxThermal;
+    thisMaterial->variableThermalAbsorptanceFunctionIdx = functionIdxThermal;
+    thisMaterial->variableSolarAbsporptanceScheduleIdx = scheduleIdxSolar;
+    thisMaterial->variableSolarAbsorptanceFunctionIdx = functionIdxSolar;
+}
+
+void GetMaterialAddOnInput(EnergyPlusData &state, bool &errorsFound)
+{
```

Comment:
This function is doing nothing except calling the other function in a loop. Splitting into two functions doesn't really make sense here.

Reply:
combined the ReadingOneMaterialAddOn into it.